libff/algebra/fields/field_utils.hpp

```diff
@@ -24,6 +24,12 @@ template<typename FieldT>
 typename std::enable_if<!std::is_same<FieldT, Double>::value, FieldT>::type
 get_root_of_unity(const size_t n);
 
+/// Decompose the input into fixed-size signed digits, of c bits. This is sometimes
+/// referred to as "fixed wnaf".
```

Comment:
In general the `w` in `wNAF` already refers to the digit size. In fact wNAF = width-w non-adjacent form. So I what you are saying here is that c=w, right?
i.e. d = \sum d_i * 2^{i}, where all |d_i| < 2^{w-1} (i.e. is w-digit long)
Right?

Reply:
Not quite.  c represents the digit size, but `d = \sum d_i * 2^{ci}`.  I.e. there is no sliding window, which I guess is where the "fixed" part comes from.  TBH, I was a bit unsure about the naming (it's not my own - I saw it used elsewhere), but used it as a succinct way to describe the decomposition.  I'll switch to something like `signed_digits` and drop the reference to wnaf.  That's probably less confusing.

Reply:
This should be renamed now.

Reply:
> TBH, I was a bit unsure about the naming (it's not my own - I saw it used elsewhere)

Do you remember in which paper you saw this?

Reply:
In the barretenberg implementation.  Although AFAICT, they have a lot of operations (including extracting the simple range of untouched bits) under the heading of wnaf. The "fixed" part kind of made sense to me (i.e. fixed rather than sliding window), but it's probably clearer to avoid it if it stands out.

Reply:
Ah right, I see